src/main/java/org/opentestsystem/ap/ims/service/impl/ItemEntityServiceImpl.java

```diff
@@ -162,4 +164,46 @@ private ItemResource buildItemResource(S3Object s3Object, String fileName) {
         }
         return itemResource;
     }
-}
+
+    private Path writeFileToDisk(S3Object s3Object, String fileName) throws IOException {
+        // account for glossary files being in a folder
+        if (StringUtils.startsWith(fileName, "glossary/")) {
+            fileName = fileName.replace("glossary/", StringUtils.EMPTY);
+        }
+        String prefix = FilenameUtils.getBaseName(fileName);
+        String suffix = "." + FilenameUtils.getExtension(fileName);
+
+        Path tempFilePath = Files.createTempFile(
+            Paths.get(this.itemBankProperties.getLocalBaseDir()), prefix, suffix);
+
+        FileUtils.copyInputStreamToFile(s3Object.getObjectContent(), tempFilePath.toFile());
+
+        return tempFilePath;
+    }
+
+    private String determineMediaType(Path filePath, String fileName) throws IOException {
+        Metadata metadata = new Metadata();
+        metadata.set(Metadata.RESOURCE_NAME_KEY, fileName);
+
+        // first look for our custom ones
+        String mediaType = this.determineMediaType(
+            new Tika(new NameDetector(this.customMediaTypes)),
```

Comment:
Hm... does instantiating a Tika consume resources?  Is it something thread-safe that we only need one of?

Reply:
I made it a member of the class.  I think it is thread-safe.